src/mmfatl.c

```diff
@@ -0,0 +1,1591 @@
+/*****************************************************************************/
+/*        Copyright (C) 2022  Wolf Lammen                                    */
+/*            License terms:  GNU General Public License                     */
+/*****************************************************************************/
+/*34567890123456 (79-character line to adjust editor window) 2345678901234567*/
+
+/*!
+ * \file mmfatl.c a self-contained set of text printing routines using
+ * dedicated pre-allocated memory, designed to likely run even under difficult
+ * conditions (corrupt state, out of memory).
+ */
+
+#include "mmfatl.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <string.h>
+
+#ifdef TEST_MMFATL
+    // We intercept malloc/free to test proper
+    // freeing after allocation of memory
+    static void* MALLOC(size_t size);
+    static void FREE(void* ptr);
+#else
+#   define FREE free
+#   define MALLOC malloc
+#endif
+
+/*!
+ * C has no boolean type, instead an int is used.  To ease code reading we
+ * encode one boolean value here.
+ * int cond = FALSE; if (cond) { } skips the code block.
+ */
+#define FALSE 0
+/*!
+ * C has no boolean type, instead an int is used.  To ease code reading we
+ * encode one boolean value here.
+ * int cond = TRUE; if (cond) { } branches into the code block.
+ */
+#define TRUE 1
+
+/*! terminating character of a text string never used elsewhere */
+#define NUL '\x00'
```

Comment:
I don't think this is preferable to just using `0`. It's only used once anyway.

Reply:
My habit